### PR TITLE
[scalardb-cluster] Support TLS for ServiceMonitor in ScalarDB Cluster chart

### PR DESCRIPTION
**Pull request by kota2and3kan**
## Description

This PR updates the `ServiceMonitor` resource to support the TLS connection from Prometheus to ScalarDB Cluster.

After this update, we can run the scrape requests from Prometheus with TLS.

Please take a look!

## Related issues and/or PRs

- This PR depends on the following PR.
  - https://github.com/scalar-labs/helm-charts/pull/263
- You can see the document about this new feature in the following PR.
  - https://github.com/scalar-labs/docs-internal-orchestration/pull/25

## Changes made

- Add new custom value `scalardbCluster.tls.caRootCertSecretForServiceMonitor` to specify the secret resource name that includes root CA certificate of ScalarDB Cluster.
- Update `charts/scalardb-cluster/templates/scalardb-cluster/servicemonitor.yaml` to enable TLS configurations when we set `scalardbCluster.tls.enabled=true`.

## Checklist

- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes.
- [x] Any remaining open issues linked to this PR are documented and up-to-date (Jira, GitHub, etc.).
- [x] Tests (unit, integration, etc.) have been added for the changes.
- [x] My changes generate no new warnings.
- [x] Any dependent changes in other PRs have been merged and published.

## Additional notes (optional)

N/A

## Release notes

Support TLS in the scrape requests from Prometheus in the ScalarDB Cluster chart.

### CODE REVIEW

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

File: charts/scalardb-cluster/README.md [hunk 1]
```diff
@@ -66,6 +66,7 @@ Current chart version is `2.0.0-SNAPSHOT`
 | scalardbCluster.strategy.rollingUpdate.maxUnavailable | string | `"25%"` | The number of pods that can be unavailable during the update process |
 | scalardbCluster.strategy.type | string | `"RollingUpdate"` | New pods are added gradually, and old pods are terminated gradually, e.g: Recreate or RollingUpdate |
 | scalardbCluster.tls.caRootCertSecret | string | `""` | Name of the Secret containing the custom CA root certificate for TLS communication. |
+| scalardbCluster.tls.caRootCertSecretForServiceMonitor | string | `""` | Name of the Secret containing the CA root certificate for TLS communication on the metrics endpoint. Prometheus Operator retrieve the CA root certificate file from this secret resource. You must create this secret resource in the same namespace as Prometheus. |
```
Comment (komamitsu) [inline charts/scalardb-cluster/README.md:69]:
```suggestion
| scalardbCluster.tls.caRootCertSecretForServiceMonitor | string | `""` | Name of the Secret containing the CA root certificate for TLS communication on the metrics endpoint. Prometheus Operator retrieves the CA root certificate file from this secret resource. You must create this secret resource in the same namespace as Prometheus. |
```

File: charts/scalardb-cluster/values.yaml [hunk 2]
```diff
@@ -284,6 +284,8 @@ scalardbCluster:
     certChainSecret: ""
     # -- Name of the Secret containing the private key file used for TLS communication.
     privateKeySecret: ""
+    # -- Name of the Secret containing the CA root certificate for TLS communication on the metrics endpoint. Prometheus Operator retrieve the CA root certificate file from this secret resource. You must create this secret resource in the same namespace as Prometheus.
```
Comment (komamitsu) [inline charts/scalardb-cluster/values.yaml:287]:
```suggestion
    # -- Name of the Secret containing the CA root certificate for TLS communication on the metrics endpoint. Prometheus Operator retrieves the CA root certificate file from this secret resource. You must create this secret resource in the same namespace as Prometheus.
```

**Review by komamitsu — APPROVED** [review]
LGTM! 👍 

Left a few minor corrections, though.

**Review by brfrn169 — APPROVED** [review]
LGTM! Thank you!

**Review by superbrothers — APPROVED** [review]

File: charts/scalardb-cluster/README.md (hunk 1 above)
Comment (kota2and3kan) [reply charts/scalardb-cluster/README.md:69]: Good catch! Thank you!

**Pull request merged**